src/main/java/io/stargate/sgv2/jsonapi/service/embedding/gateway/EmbeddingGatewayClient.java

```diff
@@ -0,0 +1,162 @@
+   * @param baseUrl - base url of the embedding client
+   * @param apiKey - Api key for the embedding provider
+   * @param modelName - Model name for the embedding provider
+   * @param embeddingService - Embedding service client
+   * @param vectorizeServiceParameter - Additional parameters for the vectorize service
+   */
+  public EmbeddingGatewayClient(
+      EmbeddingProviderConfigStore.RequestProperties requestProperties,
+      String provider,
+      int dimension,
+      Optional<String> tenant,
+      String baseUrl,
+      String apiKey,
+      String modelName,
+      EmbeddingService embeddingService,
+      Map<String, Object> vectorizeServiceParameter) {
+    this.requestProperties = requestProperties;
+    this.provider = provider;
+    this.dimension = dimension;
+    this.tenant = tenant;
+    this.apiKey = apiKey;
+    this.modelName = modelName;
+    this.baseUrl = baseUrl;
+    this.embeddingService = embeddingService;
+    this.vectorizeServiceParameter = vectorizeServiceParameter;
+  }
+
+  /**
+   * Vectorize the given list of texts
+   *
+   * @param texts List of texts to be vectorized
+   * @param apiKeyOverride API key sent as header
+   * @param embeddingRequestType Type of request (INDEX or SEARCH)
+   * @return
+   */
+  @Override
+  public Uni<List<float[]>> vectorize(
+      List<String> texts,
+      Optional<String> apiKeyOverride,
+      EmbeddingRequestType embeddingRequestType) {
+    Map<String, EmbeddingGateway.ProviderEmbedRequest.EmbeddingRequest.ParameterValue>
+        grpcVectorizeServiceParameter = new HashMap<>();
+    if (vectorizeServiceParameter != null) {
+      vectorizeServiceParameter.forEach(
+          (key, value) -> {
+            if (value instanceof String)
+              grpcVectorizeServiceParameter.put(
+                  key,
+                  EmbeddingGateway.ProviderEmbedRequest.EmbeddingRequest.ParameterValue.newBuilder()
+                      .setStrValue((String) value)
+                      .build());
+            else if (value instanceof Integer)
+              grpcVectorizeServiceParameter.put(
+                  key,
+                  EmbeddingGateway.ProviderEmbedRequest.EmbeddingRequest.ParameterValue.newBuilder()
+                      .setIntValue((Integer) value)
+                      .build());
+            else if (value instanceof Float)
+              grpcVectorizeServiceParameter.put(
+                  key,
+                  EmbeddingGateway.ProviderEmbedRequest.EmbeddingRequest.ParameterValue.newBuilder()
+                      .setFloatValue((Float) value)
+                      .build());
+            else if (value instanceof Boolean)
+              grpcVectorizeServiceParameter.put(
+                  key,
+                  EmbeddingGateway.ProviderEmbedRequest.EmbeddingRequest.ParameterValue.newBuilder()
+                      .setBoolValue((Boolean) value)
+                      .build());
+          });
```

Comment:
should also have one more `else` to throw exception for unrecognized `value` type?